### PR TITLE
Circuit for opcode `EXTCODESIZE`

### PR DESCRIPTION
**Pull request by silathdiir**
Related issue https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/993

### Summary

1. Update both `EXTCODESIZE` bus-mapping and circuit, and add test cases.

2. Fix to convert all bytecodes of bus-mapping code DB into circuit Block (not only call context bytecodes) in `block_convert` function. Since it causes account bytecode is failed to lookup in `bytecode_table` via `bytecode_length` function.
TODO: may update to the same method of `EXTCODECOPY` https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/720#issuecomment-1364631519

### CODE REVIEW

File: bus-mapping/src/evm/opcodes/extcodesize.rs [hunk 1]
```diff
@@ -12,34 +12,279 @@ impl Opcode for Extcodesize {
...
+        });
+    }
+
+    #[test]
+    fn extcodesize_opcode_non_existing_account() {
+        test_ok(None, false);
+    }
+
+    #[test]
+    fn extcodesize_opcode_empty_account() {
```
Comment (DreamWuGit) [inline bus-mapping/src/evm/opcodes/extcodesize.rs:122]: I think we can put these tests into one like: 
```
fn extcodesize_opcode_account() {
  // test empty account 
  test_ok(Some(Account::default()), false);
  // test cold account access
  test_ok(TEST_ACCOUNT.clone(), false);
  // test warm account access
   test_ok(TEST_ACCOUNT.clone(), true);
}

```

File: zkevm-circuits/src/evm_circuit/execution/extcodesize.rs [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+            address: *TEST_ADDRESS,
+            code: TEST_CODE.clone(),
+            ..Default::default()
+        });
+    }
+
+    #[test]
+    fn extcodesize_gadget_non_existing_account() {
+        test_root_ok(&None, false);
+        test_internal_ok(0x20, 0x00, &None, false);
```
Comment (DreamWuGit) [inline zkevm-circuits/src/evm_circuit/execution/extcodesize.rs:184]: for root & internal call the extcodeside follows same handling , we can only keep one , only internal call test?

File: bus-mapping/src/evm/opcodes/extcodesize.rs (hunk 1 above)
Comment (silathdiir) [reply bus-mapping/src/evm/opcodes/extcodesize.rs]: Updated and merged test cases in bus-mapping as:
```
#[test]
fn test_extcodesize_opcode() {
    // Test for non existing account.
    test_ok(None, false);
    // Test for empty account.
    test_ok(Some(Account::default()), false);
    // Test for cold account.
    test_ok(TEST_ACCOUNT.clone(), false);
    // Test for warm account.
    test_ok(TEST_ACCOUNT.clone(), true);
}
```
And also merged the test cases in zkevm-circuit as:
```
    #[test]
    fn test_extcodesize_gadget() {
        // Test for non existing account.
        test_ok(None, false);
        // Test for empty account.
        test_ok(Some(Account::default()), false);
        // Test for cold account.
        test_ok(TEST_ACCOUNT.clone(), false);
        // Test for warm account.
        test_ok(TEST_ACCOUNT.clone(), true);
    }
```

File: zkevm-circuits/src/evm_circuit/execution/extcodesize.rs (hunk 2 above)
Comment (silathdiir) [reply zkevm-circuits/src/evm_circuit/execution/extcodesize.rs]: Deleted `test_root_ok` and rename `test_internal_ok` to `test_ok`. And merged these test cases in zkevm-circuit as:
```
    #[test]
    fn test_extcodesize_gadget() {
        // Test for non existing account.
        test_ok(None, false);
        // Test for empty account.
        test_ok(Some(Account::default()), false);
        // Test for cold account.
        test_ok(TEST_ACCOUNT.clone(), false);
        // Test for warm account.
        test_ok(TEST_ACCOUNT.clone(), true);
    }
```

**Comment by silathdiir** [comment]
Thanks for dream's help review. Moved to upstream https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1014

**Pull request closed without merging**